t/re-split.t

```diff
@@ -0,0 +1,678 @@
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("a;,b;,c;,d;e", ";,")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+a
+b
+c
+d;e
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 3: split matches, no submatch, jit compile, regex cache
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("a,b,c,d", ",", "jo")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+a
+b
+c
+d
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 4: split matches + submatch (matching)
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("a;,b;,c;,d,e", "(;),")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+a
+;
+b
+;
+c
+;
+d,e
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 5: split matches + submatch (not matching)
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("a,b,c,d,e", "(;)|,")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+a
+b
+c
+d
+e
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 6: split matches + max limiter
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("a,b,c,d,e", ",", nil, nil, 3)
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+a
+b
+c
```

Comment:
This is JavaScript's `split` behavior. I tend to follow Perl's split semantics here:

```console
$ perl -e 'print join "\n", split /,/, "a,b,c,d,e", 3'
a
b
c,d,e
```